app/models/user.rb

```diff
@@ -1,11 +1,20 @@
 class User < ActiveRecord::Base
   has_secure_password validations: false
-  has_many :reviews
+  has_many :reviews, -> { order("created_at DESC") }
   has_many :queue_items, -> { order(:list_order) }
+  has_many :following_relationships, class_name: 'Relationship', foreign_key: :follower_id
+  has_many :leading_relationships, class_name: 'Relationship', foreign_key: :leader_id
 
   validates_presence_of :email, :password, :full_name
   validates_uniqueness_of :email
 
+  def can_follow?(another_user)
+    self != another_user && !follows?(another_user)
+  end
+
+  def follows?(another_user)
+    following_relationships.map(&:leader).include?(another_user)
```

Comment:
`pluck` creates an array of the chosen attribute, so it would do exactly your first scenario :-)